frontend/lindsay-wildlife-treasure-hunt-frontend/src/Components/GameDisplay/ManualEntryDisplay.js

```diff
@@ -46,66 +44,93 @@ export class ManualEntryDisplay extends React.Component {
 
 // Manual Entry Display Components
 
-class ManualEntryField extends React.Component {
+class ManualEntryNumpad extends React.Component {
   constructor(props) {
-    super(props);
+    super(props)
+    this.state = {
+      displayValue: "",
+      displayIsFull: false
+    }
+  }
 
-    this.formPreventDefault = this.formPreventDefault.bind(this);
-    this.onClickPreventDefault = this.onClickPreventDefault.bind(this);
+  digitPressed(digit) {
+    if(!this.state.displayIsFull) {
+      this.setState({
+        displayValue: this.state.displayValue + digit
+      })
+      if(this.state.displayValue.length == 1) {
+        this.setState({
+          displayIsFull: true
+        })
+      }
+    }
   }
 
-  formPreventDefault(e) {
-    e.preventDefault();
-    this.props.onClick();
+  backspace() {
+    this.setState({
+      displayValue: this.state.displayValue.substring(0, this.state.displayValue.length - 1),
+      displayIsFull: false
+    })
   }
 
-  onClickPreventDefault(e) {
-    e.preventDefault();
-    this.props.onClick();
+  handleClick(e) {
+    if(e['target']['className'].includes("num-btn")) {
```

Comment:
Can we have two separate onClick handlers for backspace, number button press, and submit? I don't really like the pattern of checking the event classNames

Reply:
Yes, I can do that.